tower-rate-limit/src/rate.rs

```diff
@@ -0,0 +1,29 @@
+use std::time::Duration;
+
+#[derive(Debug, Copy, Clone)]
+pub struct Rate {
+    num: u64,
+    per: Duration,
+}
+
+impl Rate {
+    /// Create a new rate
+    ///
+    /// # Panics
+    ///
+    /// This function panics if `num` or `per` is 0.
+    pub fn new(num: u64, per: Duration) -> Self {
+        assert!(num > 0);
+        assert!(per > Duration::from_millis(0));
+
+        Rate { num, per }
+    }
+
+    pub(crate) fn num(&self) -> u64 {
```

Comment:
do we want to remove the `pub(crate)` and have this be `pub`? I could see the use case where you might create a rate and want to access what its set too.

Reply:
I *think* if we enable the `unreachable_pub` lint, it will complain if it is not `pub(crate)`.